src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -290,30 +328,30 @@ private LanguageExpression ConvertResourcePropertyAccess(ResourceReferenceExpres
                                         },
                                         Array.Empty<LanguageExpression>())
                                 },
-                                Array.Empty<LanguageExpression>());
+                                Array.Empty<LanguageExpression>()),
+                            true);
                     case "properties":
                         // use the reference() overload without "full" to generate a shorter expression
                         // this is dependent on the name expression which could involve locals in case of a resource collection
-                        return GetReferenceExpression(resource, indexContext, false);
+                        return (GetReferenceExpression(resource, indexContext, false), true);
```

Comment:
Does this mean that `res.?properties` won't actually work? If so, one option is to change to using `Full` references here.